### ISSUE TITLE
Improve tests with inline failure messages

### ISSUE THREAD

**Issue by ZevEisenberg** [realm/SwiftLint#3035]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When developing rules and fixing bugs, trying to correlate the test output with the code can be difficult. It would be nicer if incorrectly triggered or ignored examples were highlighted in Xcode like other test failures.

This would be possible with the technique I describe in this blog post: [XCTest Line by Line](https://www.rightpoint.com/rplabs/2019/03/xctest-line-by-line/)

As a quick sketch, it would transform the test case declarations from this:

```swift
public static let description = RuleDescription(
    identifier: "foo",
    name: "Foo",
    description: "some description",
    kind: .idiomatic,
    minSwiftVersion: .fourDotOne,
    nonTriggeringExamples: [
        """
        enum SomeExample { }
        """, // etc
```

to something more like this:

```swift
public static let description = RuleDescription(
    identifier: "foo",
    name: "Foo",
    description: "some description",
    kind: .idiomatic,
    minSwiftVersion: .fourDotOne,
    nonTriggeringExamples: [
        Example("""
        enum SomeExample {}
        """), // etc etc
```

The addition of the `Example` wrapper struct would capture the file and line number, which would be passed to the `XCTest` methods, ensuring that failures are reflected on the correct line. The initializer of the `Example` struct would have to look like this:

```swift
// "example" parameter could also be a StaticString if you want to encourage
// passing string literals to aid in debugging
init(_ example: String, file: StaticString = #file, line: Uint = #line)
```

I'd be interested in contributing a PR to address this.

**Comment by jpsim** [comment]
We actually already have this, although it might have stopped working a while back.

**Comment by ZevEisenberg** [comment]
Ohh, interesting. Do you know where it is/was? I can look into fixing it up. It would be a real quality-of-life improvement for people doing small fixes who aren't as familiar with the code base as a whole.

**Comment by jpsim** [comment]
Turns out we do this for the integration tests: https://github.com/realm/SwiftLint/blob/0.38.2/Tests/SwiftLintFrameworkTests/IntegrationTests.swift#L26-L28

But we don't do this for the rule examples. This would be a lovely addition.

**Comment by ZevEisenberg** [comment]
I'm already working on it, but it turns out there are a _ton_ of rules with weirdly formatted strings, which makes some of this a little tricky. Thoughts on converting them all to multiline strings as part of the conversion effort? I'm pretty sure that either way, this will wreak havoc on any open PRs that substantially change any examples :\

I had a cool idea of using `ExpressibleByStringLiteral` to provide a compatibility mode for contributors, which would fail the tests (instead of failing the build) until they've wrapped their examples in `Example(...)`, but that doesn't work for the hundreds or thousands of tests that use old-style `"foo\n" + "bar"` string concatenations.

That's why I'm currently investigating [SwiftRewriter](https://github.com/inamiy/SwiftRewriter) to see if it's worth building some kind of utility to do the conversion to multiline strings, but the answer is almost definitely "it is not worth it."

**Comment by ZevEisenberg** [comment]
Do you know why so many triggering/non-triggering examples in SwiftLint have a `\n` at the end of them? Is it required? Wondering if I should clean those up as part of this work. It would certainly clear up some visual noise in the test strings.
